docroot/modules/custom/va_gov_vet_center/src/Plugin/QueueWorker/RequiredServiceBase.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\va_gov_vet_center\Plugin\QueueWorker;
+
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\node\Entity\Node;
+
+/**
+ * Provides base functionality for the Subscription Builder Queue Workers.
+ */
+abstract class RequiredServiceBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $nodeStorage;
+
+  /**
+   * The term storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $termStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    EntityStorageInterface $node_storage,
+    EntityStorageInterface $term_storage
+  ) {
+    $this->nodeStorage = $node_storage;
+    $this->termStorage = $term_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $container->get('entity_type.manager')->getStorage('node'),
+      $container->get('entity_type.manager')->getStorage('taxonomy_term')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data) {
+    // Use the "cms migrator" user account (1317).
+    $author_uid = 1317;
+    $vet_center = $this->nodeStorage->load($data->facility_id);
+    $published = $vet_center->isPublished();
+    $section_id = $vet_center->field_administration->target_id;
+    $moderation_state = ($published ? 'published' : 'draft');
+
+    // Create the new node.
+    $node = Node::create([
```

Comment:
We've got a couple of nodes we are juggling in this function.   Lets be specific about what kind of node it is to avoid confusion.
```suggestion
    $vet_center_facility_health_service_node = Node::create([
```